### PR TITLE
Support Puppet v4 AIO

### PR DESCRIPTION
**Pull request by ghoneycutt**

### CODE REVIEW

**Comment by ghoneycutt** [comment]
@tphoney Could you please take a look

File: metadata.json [hunk 1]
```diff
@@ -70,11 +70,15 @@
   ],
   "requirements": [
     {
+      "name": "pe",
+      "version_requirement": ">= 3.2.0 < 5.0.0"
+    },
```
Comment (DavidS) [inline metadata.json]: This is not needed anymore. the forge evaluates that from the `puppet` requirement.

File: Rakefile [hunk 2]
```diff
@@ -1,11 +1,19 @@
...
-
-PuppetLint.configuration.fail_on_warnings = true
-PuppetLint.configuration.send('relative')
 PuppetLint.configuration.send('disable_80chars')
 PuppetLint.configuration.send('disable_140chars')
-PuppetLint.configuration.send('disable_class_inherits_from_params_class')
-PuppetLint.configuration.send('disable_documentation')
-PuppetLint.configuration.send('disable_single_quote_string_with_variables')
+PuppetLint.configuration.relative = true
 PuppetLint.configuration.ignore_paths = ["spec/**/*.pp", "pkg/**/*.pp"]
```
Comment (DavidS) [inline Rakefile]: Current versions of PSH provide better defaults for this.

File: Rakefile [hunk 3]
```diff
@@ -1,11 +1,19 @@
...
+  Dir['manifests/**/*.pp'].each do |manifest|
+    sh "puppet parser validate --noop #{manifest}"
+  end
+  Dir['spec/**/*.rb','lib/**/*.rb'].each do |ruby_file|
+    sh "ruby -c #{ruby_file}" unless ruby_file =~ /spec\/fixtures/
+  end
+  Dir['templates/**/*.erb'].each do |template|
+    sh "erb -P -x -T '-' #{template} | ruby -c"
+  end
+end
```
Comment (DavidS) [inline Rakefile]: Current versions of PSH provide this.

**Comment by tphoney** [comment]
i agree with @DavidS good work @ghoneycutt !!

File: Rakefile (hunk 3 above)
Comment (ghoneycutt) [inline Rakefile]: It only tests `lib/*`. It is also undocumented what it does.

https://github.com/puppetlabs/puppetlabs_spec_helper/blob/6c8aa5b921f96e860b0f5f92dde13647572122ff/lib/puppetlabs_spec_helper/rake_tasks.rb#L450

File: metadata.json (hunk 1 above)
Comment (ghoneycutt) [inline metadata.json]: Shall I rebase or can this be merged?

**Comment by tphoney** [comment]
merging, it would  be great if you could address the issues outlined

thanks for the continued work.

**Comment by ghoneycutt** [comment]
Thank you for merging. Can you please tag a new release. I'll update the metadata in another PR now.

**Comment by tphoney** [comment]
@ghoneycutt i have created a ticket for us to do a new release in an upcoming sprint.

**Pull request merged**